arduino-ide-extension/src/browser/boards/boards-service-provider.ts

```diff
@@ -409,14 +409,14 @@ export class BoardsServiceProvider
   }
 
   async selectedBoardUserFields(): Promise<BoardUserField[]> {
-    if (!this._boardsConfig.selectedBoard || !this._boardsConfig.selectedPort) {
+    if (!this._boardsConfig.selectedBoard) {
       return [];
     }
     const fqbn = this._boardsConfig.selectedBoard.fqbn;
     if (!fqbn) {
       return [];
     }
-    const protocol = this._boardsConfig.selectedPort.protocol;
+    const protocol = this._boardsConfig.selectedPort?.protocol || 'default';
```

Comment:
Please link the [doc](https://arduino.github.io/arduino-cli/dev/platform-specification/#sketch-upload-configuration) or a single-line comment about why `'default'` is OK here? Otherwise, we will see `|| 'default'` everywhere in the code pretty soon. What do you think?

Reply:
Good idea. I have done it here:
https://github.com/arduino/arduino-ide/compare/79e431535a884e434e7efd000714e455697e7835..b76e7f99f0d7c33f48097b27e3fb487cb9cc582e